pkgs/development/python-modules/torchvision/default.nix

```diff
@@ -34,24 +52,29 @@ in buildPythonPackage rec {
   };
 
   nativeBuildInputs = [ libpng ninja which ]
-    ++ lib.optionals cudaSupport [ cudatoolkit_joined ];
-
-  TORCHVISION_INCLUDE = "${libjpeg_turbo.dev}/include/";
-  TORCHVISION_LIBRARY = "${libjpeg_turbo}/lib/";
+    ++ lib.optional cudaSupport cuda-native-redist;
 
   buildInputs = [ libjpeg_turbo libpng ]
-    ++ lib.optionals cudaSupport [ cudnn ];
+    ++ lib.optional cudaSupport cuda-redist;
 
   propagatedBuildInputs = [ numpy pillow torch scipy ];
 
-  preBuild = lib.optionalString cudaSupport ''
+  # NOTE: We essentially override the compilers provided by stdenv because we don't have a hook
+  #   for cudaPackages to swap in compilers supported by NVCC.
+  preConfigure = lib.optionalString cudaSupport ''
+    export CC=${cudatoolkit.cc}/bin/cc
+    export CXX=${cudatoolkit.cc}/bin/c++
+    export CUDAHOSTCXX=${cudatoolkit.cc}/bin/c++
+    export TORCHVISION_INCLUDE="${libjpeg_turbo.dev}/include/"
+    export TORCHVISION_LIBRARY="${libjpeg_turbo}/lib/"
```

Comment:
are these variables only necessary when `cudaSupport = true`? this looks to change them to only being set when `cudaSupport = true`

Reply:
You're right, fixed in the new version